### DISCUSSION TITLE
Installation PowerShell mit "Install-PowerShell" unter Windows 7 Pro

### DISCUSSION THREAD

**Discussion by helmutkilian** [TobiasPSP/OReilly#6] [Allgemeines]
Buch S. 26-28:
Das funktioniert unter Windows 7 zunächst einmal **nicht**, weil das Skript 
https://aka.ms/install-powershell.ps1 
unter "Windows PowerShell" nicht durchläuft. Schuld daran ist der Befehl "Get-ComputerInfo". Dieser scheitert mit der Fehlermeldung
    Der Einstiegspunkt "GetFirmwareType" wurde nicht in der DLL "kernel32.dll" gefunden.

Die Installation gelingt, wenn man sich das Skript kopiert und ein klein wenig verändert. "Get-ComputerInfo" wird nämlich nur benutzt, um die Prozessorarchitektur ARM zu ermitteln. Da man die Architektur vom eigenen Rechner ja kennt oder ermitteln kann, braucht man "Get-ComputerInfo" nicht. Ich habe kein Windows auf ARM-Prozessor und musste deshalb nur zwei Zeilen auskommentieren.

Mit dem angepassten Skript erzeugt man nun den Befehl "Install-PowerShell", und es funktioniert.

**Comment by TobiasPSP** [comment]
Sehr guter Hinweis. `Get-ComputerInfo` wurde mit PowerShell 5 nachgeliefert und setzt auf API-Schnittstellen, die in älteren Windows-Versionen fehlen. Daher funktioniert dieses Cmdlet bei älteren Windows-Versionen nicht richtig. Da es aber genau wie von Ihnen beschrieben gar nicht für die Installation erforderlich ist, sondern nur die Rahmenbedingungen erforscht, unter denen danach die Installation passieren soll, ist Ihr Workaround ein sehr guter Weg,
